homeworks/week6/trang_vu6.java

```diff
@@ -0,0 +1,65 @@
+ 
+ class trang_vu6{
+  public static void main(String[] args) {
+    System.out.println(getShortestSubstringOfAllCharacters("xyyzyzyx", "xyz"));  // "zyx"
+  }
+
+  // Time complexity O(n)
+  // Space complexity O(n)
```

Comment:
`n` is not being defined. If `n = str.length`, then it is incorrect. O(M) is the correct answer where M is pat.length.